packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FileChooserActivity.java

```diff
@@ -0,0 +1,219 @@
+    Intent filesIntent = new Intent(Intent.ACTION_GET_CONTENT);
+
+    if (getIntent().getBooleanExtra(EXTRA_ALLOW_MULTIPLE_FILES, false)) {
+      filesIntent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true);
+    }
+
+    String[] acceptTypes = getIntent().getStringArrayExtra(EXTRA_ACCEPT_TYPES);
+
+    if (acceptTypes.length == 0 || (acceptTypes.length == 1 && acceptTypes[0].length() == 0)) {
+      // empty array or only 1 empty string? -> accept all types
+      filesIntent.setType("*/*");
+    } else if (acceptTypes.length == 1) {
+      filesIntent.setType(acceptTypes[0]);
+    } else {
+      // acceptTypes.length > 1
+      filesIntent.setType("*/*");
+      filesIntent.putExtra(Intent.EXTRA_MIME_TYPES, acceptTypes);
+    }
+
+    return (filesIntent.resolveActivity(getPackageManager()) != null) ? filesIntent : null;
+  }
+
+  private Intent createCaptureIntent(String type, String fileFormat) {
+    Intent captureIntent = new Intent(type);
+    if (captureIntent.resolveActivity(getPackageManager()) == null) {
+      return null;
+    }
+
+    // Create the File where the output should go
+    Uri captureOutputUri = getTempUri(fileFormat);
+    potentialCaptureOutputUris.add(captureOutputUri);
+
+    captureIntent.putExtra(MediaStore.EXTRA_OUTPUT, captureOutputUri);
+
+    return captureIntent;
+  }
+
+  private File getStorageDirectory() {
+    File imageDirectory = new File(getCacheDir(), WEBVIEW_STORAGE_DIRECTORY);
+    if (!imageDirectory.exists() && !imageDirectory.mkdir()) {
+      Log.e("WEBVIEW", "Unable to create storage directory");
+    }
+    return imageDirectory;
+  }
+
+  private Uri getTempUri(String format) {
+    String fileName = "CAPTURE-" + simpleDateFormat.format(new Date()) + "." + format;
+    File file = new File(getStorageDirectory(), fileName);
+    return FileProvider.getUriForFile(
+        this, getApplicationContext().getPackageName() + ".generic.provider", file);
+  }
+
+  private String getFileNameFromUri(Uri uri) {
+    Cursor returnCursor = getContentResolver().query(uri, null, null, null, null);
+    assert returnCursor != null;
+    int nameIndex = returnCursor.getColumnIndex(OpenableColumns.DISPLAY_NAME);
+    returnCursor.moveToFirst();
+    String name = returnCursor.getString(nameIndex);
+    returnCursor.close();
+    return name;
+  }
+
+  private Uri copyToLocalUri(Uri uri) {
+    File destination = new File(getStorageDirectory(), getFileNameFromUri(uri));
+
+    try (InputStream in = getContentResolver().openInputStream(uri);
+        OutputStream out = new FileOutputStream(destination)) {
+      byte[] buffer = new byte[1024];
+      int len;
+      while ((len = in.read(buffer)) != -1) {
+        out.write(buffer, 0, len);
+      }
+      return FileProvider.getUriForFile(
+          this, getApplicationContext().getPackageName() + ".generic.provider", destination);
+    } catch (IOException e) {
+      Log.e("WEBVIEW", "Unable to copy selected image", e);
```

Comment:
Use a private class field for the tag:

```java
public class FileChooserActivity extends Activity {
  private static final String TAG = "FileChooserActivity";
  
  ...
  
  private File getStorageDirectory() {
    ...
    if (!imageDirectory.exists() && !imageDirectory.mkdir()) {
      Log.e(TAG, "Unable to create storage directory");
    }
    ...
  }

  ...

  private Uri copyToLocalUri(Uri uri) {
    try (InputStream in = getContentResolver().openInputStream(uri);
         OutputStream out = new FileOutputStream(destination)) {
      ...
    } catch (IOException e) {
      Log.e(TAG, "Unable to copy selected image", e);
      ...
    }
  }

}
```

Reply:
Thanks for your comment. I've added this change.